server/src/main/java/io/crate/metadata/bugfix/CorruptedMetadataFixer.java

```diff
@@ -0,0 +1,222 @@
+                    generateTemplateIndexMetadata(indexName, indexMetadata, mappingMetadata).build());
+
+                fixedMetadata.removeTemplate(templateName);
+                fixedMetadata.templates(mapBuilder.build());
+            } else {
+                /*
+                   ex) test_swap_table_partitioned_to_non_partitioned_3
+                   - index:
+                       "m7.t7" - not partitioned / contains column s
+                       "m7.s7" - partitioned by t
+                   - templates:
+                       "m7..partitioned.t7." - partitioned by t
+                   > since "m7.t7" is not partitioned, "m7..partitioned.t7." can be dropped.
+                */
+                fixedMetadata.removeTemplate(PartitionName.templateName(indexParts[0], indexParts[1]));
+            }
+        }
+    }
+
+    /**
+     * Generates an indexTemplateMetadata based on the given indexMetadata.
+     */
+    private static IndexTemplateMetadata.Builder generateTemplateIndexMetadata(String indexName,
+                                                                               IndexMetadata indexMetadata,
+                                                                               MappingMetadata mappingMetadata) {
+        String[] indexParts = indexName.split("\\.");
+
+        String templateName = PartitionName.templateName(indexParts[0], indexParts[1]);
+        IndexTemplateMetadata.Builder templateBuilder = new IndexTemplateMetadata.Builder(templateName);
+        templateBuilder.putAlias(AliasMetadata.builder(indexName).build());
+        templateBuilder.putMapping(mappingMetadata.source());
+        Settings indexSettings = indexMetadata.getSettings();
+        var templateCompatibleSettings =
+            // these settings cause exceptions when added to templates
+            indexSettings.filter(k -> !(k.equals(SETTING_INDEX_UUID) || k.equals(SETTING_CREATION_DATE)));
+
+        templateBuilder.settings(templateCompatibleSettings);
+        templateBuilder.patterns(List.of(PartitionName.templatePrefix(indexParts[0], indexParts[1])));
+
+        return templateBuilder;
+    }
+
+    /**
+     * Generates a partitioned indexMetadata based on the given indexMetadata.
+     */
+    private static IndexMetadata.Builder generatePartitionedIndexMetadata(String indexName,
+                                                                          IndexMetadata indexMetadata,
+                                                                          List<List<String>> partitionedByColumns) {
+        String[] indexParts = indexName.split("\\.");
+
+        IndexMetadata.Builder partitionedIndexMetadata = new IndexMetadata.Builder(indexMetadata);
+        partitionedIndexMetadata.putAlias(AliasMetadata.builder(indexName));
+        String partitionedIndexName = new PartitionName(
+            new RelationName(indexParts[0], indexParts[1]),
+            partitionedByColumns.stream().map(c -> (String) null).toList())
```

Comment:
We set the lost `partitioned_by` columns to nulls. We could consider filtering `not null` constraints but the trade-off would be that it would deviate more from the original table structure.